osc-server/src/main/java/org/osc/core/broker/util/VirtualizationConnectorUtil.java

```diff
@@ -75,82 +83,19 @@ private void delayedInit() {
-                            vcDto.getProviderUser(), vcDto.getProviderPassword(), isHttps,
-                            SslContextProvider.getInstance().getSSLContext());
-
-                    try (Openstack4jKeystone keystoneAPi = new Openstack4jKeystone(endPoint)) {
-                        keystoneAPi.listProjects();
-                    }
-                } catch (Exception exception) {
-                    errorTypeException = new ErrorTypeException(exception, ErrorType.PROVIDER_EXCEPTION);
-                    LOG.warn(
-                            "Exception encountered when trying to add Keystone info to Virtualization Connector, allowing user to either ignore or correct issue");
-                }
-            }
-
-            if (!request.isIgnoreErrorsAndCommit(ErrorType.RABBITMQ_EXCEPTION)) {
-                initSSLCertificatesListener(this.managerFactory, certificateResolverModels, "rabbitmq");
-                try {
-                    OsRabbitMQClient rabbitClient = new OsRabbitMQClient(vc);
-                    rabbitClient.testConnection();
-                } catch (ShutdownSignalException shutdownException) {
-                    // If its an existing VC which we are connected to, then this exception is expected
-                    if (vc.getId() != null) {
-                        delayedInit();
-                        OsRabbitMQClient osRabbitMQClient = this.activeRunner.getVcToRabbitMQClientMap().get(vc.getId());
-                        if (osRabbitMQClient != null && osRabbitMQClient.isConnected()) {
-                            LOG.info(
-                                    "Exception encountered when connecting to RabbitMQ, ignoring since we are already connected",
-                                    shutdownException);
-                        } else {
-                            errorTypeException = new ErrorTypeException(shutdownException,
-                                    ErrorType.RABBITMQ_EXCEPTION);
-                        }
-                    } else {
-                        errorTypeException = new ErrorTypeException(shutdownException, ErrorType.RABBITMQ_EXCEPTION);
-                    }
-                } catch (Throwable exception) {
-                    errorTypeException = new ErrorTypeException(exception, ErrorType.RABBITMQ_EXCEPTION);
-                    LOG.warn(
-                            "Exception encountered when trying to connect to RabbitMQ, allowing user to either ignore or correct issue");
-                }
+            if (vc.getVirtualizationType().equals(VirtualizationType.OPENSTACK)) {
+                errorTypeException = checkOpenstackConnection(request, certificateResolverModels, vc);
+            } else if (vc.getVirtualizationType().equals(VirtualizationType.KUBERNETES)) {
+                errorTypeException = checkKubernetesConnection(request, certificateResolverModels, vc);
             }
 
             this.managerFactory.clearListener();
```

Comment:
seems odd place for this to be....move it to where the manager factory listener is added

Reply:
Can you be more specific, i.e.: which ln? This has not changed but I can move if there is no risk of regression in doing so.  As it was before there are multiple places where the listener is being initiated, all of them private methods, this being the common public method.

Reply:
i see initSSLCertificatesListener(this.managerFactory, certificateResolverModels, "rabbitmq"); method calls and the clear listener is the counterpart for it. With this refactor, i dont see where the listeners are initialized. It makes sense to keep them together(atleast in the same method) so there is clarity